### PR TITLE
Move applicable Unit Tests for commands Part 1

### PR DESCRIPTION
**Pull request by wolfylambova22**
Following HMI commands were covered by Unit Tests:

    activate_app_request
    activate_app_response
    add_statistics_info_notification
    allow_all_apps_request
    allow_all_apps_response
    allow_app_request
    allow_app_response
    basic_communication_system_request
    basic_communication_system_response
    button_get_capabilities_request
    button_get_capabilities_response
    navi_send_location_request
    navi_send_location_response
    navi_show_constant_tbt_request
    navi_show_constant_tbt_response
    navi_start_stream_request
    navi_start_stream_response
    navi_stop_stream_request
    navi_stop_stream_response
    navi_subscribe_way_points_request
    navi_subscribe_way_points_response
    navi_unsubscribe_way_points_request
    navi_unsubscribe_way_points_response
    navi_update_turn_list_request
    navi_update_turn_list_response
    sdl_activate_app_request
    sdl_activate_app_response
    sdl_get_list_of_permissions_request
    sdl_get_list_of_permissions_response
    sdl_get_status_update_request
    sdl_get_status_update_response
    
  **Additional UTs were added to sdl_activate_app_request.cc

    Related issue: APPLINK-26069

### CODE REVIEW

**Comment by wolfylambova22** [comment]
@AByzhynar @AGaliuzov @okozlovlux @LevchenkoS @anosach-luxoft @orhan-mehmedov Please review.

File: src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc [hunk 1]
```diff
@@ -213,17 +225,7 @@ TYPED_TEST(RequestToHMICommandsTest, Run_SendMessageToHMI_SUCCESS) {
 
   SharedPtr<CommandType> command = this->template CreateCommand<CommandType>();
 
-  EXPECT_CALL(this->mock_app_manager_, SendMessageToHMI(NotNull()));
-
-  command->Run();
-}
-
-TYPED_TEST(RequestToHMICommandsTest2, Run_SendMessageToHMI_SUCCESS) {
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc]: @wolfylambova22, why did you delete it?

Comment (wolfylambova22) [inline src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc]: @LevchenkoS I fixed that right after PR. I used a new mergetool and lines were deleted by mistake. It is already fixed as you will see in next commit.

**Comment by AByzhynar** [comment]
@wolfylambova22 Bad commit name.  Please check the article.  Also information was sent to all developers by email.  DON'T USE 'PASA'  in commits or PR to OpenSDL.  
Ask @vlantonov for details.

File: src/components/application_manager/test/commands/hmi/activate_app_request_test.cc [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+using am::commands::CommandImpl;
+
+typedef ::utils::SharedPtr<ActivateAppRequest> ActivateAppRequestPtr;
+
+namespace {
+const uint32_t kAppId = 1u;
+const uint32_t kCorrelationId = 2u;
+}  // namespace
+
+class ActivateAppRequestTest : public CommandsTest<CommandsTestMocks::kIsNice> {
```
Comment (orhan-mehmedov) [inline src/components/application_manager/test/commands/hmi/activate_app_request_test.cc]: @wolfylambova22 In general it is good to put such local class declarations in the above anonymous namespace. In case there is a class with same name in another source file you may confuse the linker and get strange results. I'm almost sure that this is not the case here, but I suggest it (to all) as a good practice. I solved already such an issue today in unit tests.

File: src/components/application_manager/test/commands/hmi/activate_app_request_test.cc [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    msg_params[strings::correlation_id] = kCorrelationId;
+    (*msg)[strings::msg_params] = msg_params;
+    (*msg)[strings::params][strings::app_id] = kAppId;
+    (*msg)[strings::params][strings::correlation_id] = kCorrelationId;
+    (*msg)[strings::app_id] = kAppId;
+    return msg;
+  }
+};
+
+TEST_F(ActivateAppRequestTest, Run_SUCCESS) {
```
Comment (orhan-mehmedov) [inline src/components/application_manager/test/commands/hmi/activate_app_request_test.cc]: @wolfylambova22 I see only "success" test case. Is it expected to have other test cases? 
P.S. Question is valid to all other files in the commit.

File: src/components/application_manager/src/commands/hmi/navi_start_stream_request.cc [hunk 4]
```diff
@@ -44,8 +44,10 @@ NaviStartStreamRequest::NaviStartStreamRequest(
     , EventObserver(application_manager.event_dispatcher())
     , retry_number_(0) {
   SDL_AUTO_TRACE();
+  const ApplicationManagerSettings& settings =
+      application_manager_.get_settings();
   std::pair<uint32_t, int32_t> stream_retry =
-      application_manager_.get_settings().start_stream_retry_amount();
+      settings.start_stream_retry_amount();
```
Comment (wolfylambova22) [inline src/components/application_manager/src/commands/hmi/navi_start_stream_request.cc]: @orhan-mehmedov my task is to move applicable existing UTs and create appropriate ones if existant are not applicable. This test is not written by me but I believe new variable is introduced in favor of easier code readability.

Comment (LevchenkoS) [inline src/components/application_manager/src/commands/hmi/navi_start_stream_request.cc]: @wolfylambova22, this variable used just once. I agree with @orhan-mehmedov, and so, see no reason why it should be here. Also, this change is not necessary and have no effect, nor for tests, which you trying to move, and especially, nor for realization.

File: src/components/application_manager/test/commands/hmi/dummy_hmi_commands_test.cc [hunk 5]
```diff
@@ -264,6 +263,7 @@
 #include "application_manager/commands/hmi/on_vi_steering_wheel_angle_notification.h"
 #include "application_manager/commands/hmi/on_vi_my_key_notification.h"
 #include "application_manager/commands/hmi/ui_set_icon_request.h"
+#include "application_manager/commands/hmi/navi_stop_stream_response.h"
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/hmi/dummy_hmi_commands_test.cc]: @wolfylambova22, this header already included.

File: src/components/application_manager/test/commands/hmi/allow_all_apps_response_test.cc [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+using application_manager::commands::MessageSharedPtr;
+using application_manager::commands::AllowAllAppsResponse;
+
+namespace strings = ::application_manager::strings;
+namespace hmi_response = ::application_manager::hmi_response;
+
+typedef ::utils::SharedPtr<AllowAllAppsResponse> ResponsePtr;
+
+namespace {
+const bool kIsResponseAllowed = true;
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/hmi/allow_all_apps_response_test.cc]: @wolfylambova22 `kIsResponseAllowed ` is not a good name for `const`. As it's always `true`.
I think `kResponseIsAllowed` will be better as from its name it will be clear that response is allowed and can not be changed

File: src/components/application_manager/test/commands/hmi/simple_response_from_hmi_test.cc [hunk 7]
```diff
@@ -244,7 +252,7 @@ TEST_F(OtherResponseFromHMICommandsTest, VIGetVehicleTypeResponse_Run_SUCCESS) {
       CreateCommand<commands::VIGetVehicleTypeResponse>(command_msg));
 
   application_manager_test::MockHMICapabilities hmi_capabilities;
-  EXPECT_CALL(mock_app_manager_, hmi_capabilities())
+  EXPECT_CALL(app_mngr_, hmi_capabilities())
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/hmi/simple_response_from_hmi_test.cc]: @wolfylambova22, did you try to compile SDL with turned on `am_commands_test` target?

File: src/components/application_manager/test/commands/hmi/navi_start_stream_request_test.cc [hunk 8]
```diff
@@ -0,0 +1,330 @@
...
+    (*command_msg)[am::strings::params][am::strings::correlation_id] = kCorId;
+    app = CreateMockApp();
+
+    ON_CALL(app_mngr_settings_, start_stream_retry_amount())
+        .WillByDefault(ReturnRef(stream_retry));
+  }
+
+  MockAppPtr app;
+  MessageSharedPtr command_msg;
+  std::pair<uint32_t, int32_t> stream_retry;
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/hmi/navi_start_stream_request_test.cc]: @wolfylambova22 Can't find the header for `std::pair`

File: src/components/application_manager/src/commands/hmi/navi_start_stream_request.cc [hunk 9]
```diff
@@ -44,8 +44,10 @@ NaviStartStreamRequest::NaviStartStreamRequest(
     , EventObserver(application_manager.event_dispatcher())
     , retry_number_(0) {
   SDL_AUTO_TRACE();
+  const ApplicationManagerSettings& settings =
+      application_manager_.get_settings();
   std::pair<uint32_t, int32_t> stream_retry =
```
Comment (AByzhynar) [inline src/components/application_manager/src/commands/hmi/navi_start_stream_request.cc]: @wolfylambova22 Can't find the header for `std::pair`

File: src/components/application_manager/test/commands/hmi/navi_start_stream_request_test.cc [hunk 10]
```diff
@@ -0,0 +1,330 @@
...
+namespace {
+const int32_t kConnectionKey = 2;
+const uint32_t kAppId = 3u;
+const uint32_t kCorId = 5u;
+}  // namespace
+
+class NaviStartStreamRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ protected:
+  NaviStartStreamRequestTest() {}
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/hmi/navi_start_stream_request_test.cc]: @wolfylambova22 What do you need this empty ctor for?

File: src/components/application_manager/test/commands/hmi/sdl_get_status_update_request_test.cc [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+namespace hmi_response = ::application_manager::hmi_response;
+
+class SDLGetStatusUpdateRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {};
+
+TEST_F(SDLGetStatusUpdateRequestTest, Run_SUCCESS) {
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[strings::params][strings::correlation_id] = kCorrelationID;
+
+  SharedPtr<SDLGetStatusUpdateRequest> command(
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/hmi/sdl_get_status_update_request_test.cc]: @wolfylambova22 Missing header for `SharedPtr`

File: src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc [hunk 12]
```diff
@@ -131,12 +141,12 @@ TEST_F(RequestToHMITest, SendRequest_SUCCESS) {
   SharedPtr<commands::RequestToHMI> command(
       CreateCommand<commands::RequestToHMI>());
 
-  EXPECT_CALL(mock_app_manager_, SendMessageToHMI(NotNull()));
+  EXPECT_CALL(app_mngr_, SendMessageToHMI(NotNull()));
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc]: @wolfylambova22 Do you use here real instance of application manager? If not - this change is not good

File: src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc [hunk 13]
```diff
@@ -131,12 +141,12 @@ TEST_F(RequestToHMITest, SendRequest_SUCCESS) {
...
 
   command->SendRequest();
 }
 
-template <typename Command>
+template <class Command>
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc]: @wolfylambova22 Why did you change it? We don't use `<class ...>` in templates. Please read carefully: [SDL Coding Style](https://github.com/smartdevicelink/sdl_core/wiki/SDL-Coding-Style-Guide) and make appropriate changes.

File: src/components/application_manager/test/commands/hmi/simple_response_from_hmi_test.cc [hunk 14]
```diff
@@ -112,7 +120,7 @@ class ResponseFromHMICommandsTest
   MockEventDispatcher event_dispatcher_;
 
   ResponseFromHMICommandsTest() {
-    ON_CALL(mock_app_manager_, event_dispatcher())
+    ON_CALL(app_mngr_, event_dispatcher())
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/hmi/simple_response_from_hmi_test.cc]: @wolfylambova22 Do you use here real instance of application manager? If not - this change is not good. Return it back in all places where mock used.

File: src/components/application_manager/src/commands/hmi/navi_start_stream_request.cc (hunk 9 above)
Comment (wolfylambova22) [inline src/components/application_manager/src/commands/hmi/navi_start_stream_request.cc]: @AByzhynar this variable has been declared by A.Kutsan so I dont know why he didn't include header, but ok, if you insist I'll include it.

File: src/components/application_manager/test/commands/hmi/activate_app_request_test.cc (hunk 3 above)
Comment (wolfylambova22) [inline src/components/application_manager/test/commands/hmi/activate_app_request_test.cc]: @orhan-mehmedov As I told you, I am MOVING applicable tests between repos and making changes to them in order to be applicable. Anyway, I checked, currently this test is enough.

File: src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc (hunk 1 above)
Comment (wolfylambova22) [inline src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc]: @LevchenkoS fixed in b88a96b90550752923d6114406875a2541ff641d.

File: src/components/application_manager/src/commands/hmi/navi_start_stream_request.cc (hunk 4 above)
Comment (wolfylambova22) [inline src/components/application_manager/src/commands/hmi/navi_start_stream_request.cc]: @orhan-mehmedov @LevchenkoS fixed in b88a96b90550752923d6114406875a2541ff641d.

File: src/components/application_manager/test/commands/hmi/simple_response_from_hmi_test.cc (hunk 7 above)
Comment (wolfylambova22) [inline src/components/application_manager/test/commands/hmi/simple_response_from_hmi_test.cc]: @LevchenkoS fixed in 80c444ec1c84baca57b54c1c56f22460e39aaa32.

File: src/components/application_manager/test/commands/hmi/navi_start_stream_request_test.cc (hunk 8 above)
Comment (wolfylambova22) [inline src/components/application_manager/test/commands/hmi/navi_start_stream_request_test.cc]: @AByzhynar fixed in b88a96b90550752923d6114406875a2541ff641d.

File: src/components/application_manager/src/commands/hmi/navi_start_stream_request.cc (hunk 9 above)
Comment (wolfylambova22) [inline src/components/application_manager/src/commands/hmi/navi_start_stream_request.cc]: @AByzhynar fixed in b88a96b90550752923d6114406875a2541ff641d.

File: src/components/application_manager/test/commands/hmi/navi_start_stream_request_test.cc (hunk 10 above)
Comment (wolfylambova22) [inline src/components/application_manager/test/commands/hmi/navi_start_stream_request_test.cc]: @AByzhynar fixed in b88a96b90550752923d6114406875a2541ff641d.

File: src/components/application_manager/test/commands/hmi/sdl_get_status_update_request_test.cc (hunk 11 above)
Comment (wolfylambova22) [inline src/components/application_manager/test/commands/hmi/sdl_get_status_update_request_test.cc]: @AByzhynar fixed in b88a96b90550752923d6114406875a2541ff641d.

File: src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc (hunk 12 above)
Comment (wolfylambova22) [inline src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc]: @AByzhynar fixed in b88a96b90550752923d6114406875a2541ff641d.

File: src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc (hunk 13 above)
Comment (wolfylambova22) [inline src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc]: @AByzhynar fixed in b88a96b90550752923d6114406875a2541ff641d.

File: src/components/application_manager/test/commands/hmi/simple_response_from_hmi_test.cc (hunk 14 above)
Comment (wolfylambova22) [inline src/components/application_manager/test/commands/hmi/simple_response_from_hmi_test.cc]: @AByzhynar fixed in b88a96b90550752923d6114406875a2541ff641d.

File: src/components/application_manager/test/CMakeLists.txt [hunk 15]
```diff
@@ -167,7 +167,7 @@ list(APPEND LIBRARIES
 create_test(am_hmi_capabilities_test "${AM_HMI_CAPABILITIES_SOURCES}" "${LIBRARIES}")
 create_test(am_application_test "${APPLICATION_SOURCES}" "${LIBRARIES}")
 # TODO(AKutsan) APPLINK-25547 Uncoment command tests when sdl will support multithreading build
```
Comment (LevchenkoS) [inline src/components/application_manager/test/CMakeLists.txt]: @wolfylambova22, please remove this comment, and squash this commit with previous.

**Comment by AByzhynar** [comment]
Reviewed

File: src/components/application_manager/test/commands/mobile/set_global_properties_request_test.cc [hunk 16]
```diff
@@ -59,7 +59,7 @@ using utils::custom_string::CustomString;
 using smart_objects::SmartObject;
 using testing::_;
 using testing::Return;
-
+/*
```
Comment (wolfylambova22) [inline src/components/application_manager/test/commands/mobile/set_global_properties_request_test.cc]: Commented code will be fixed in next commit.

File: src/components/application_manager/test/commands/hmi/dummy_hmi_commands_test.cc (hunk 5 above)
Comment (wolfylambova22) [inline src/components/application_manager/test/commands/hmi/dummy_hmi_commands_test.cc]: @LevchenkoS fixed in ba0bc031e6ecb39a14d5302e2158a254f25e4aba.

File: src/components/application_manager/test/commands/hmi/allow_all_apps_response_test.cc (hunk 6 above)
Comment (wolfylambova22) [inline src/components/application_manager/test/commands/hmi/allow_all_apps_response_test.cc]: @AByzhynar fixed in ba0bc031e6ecb39a14d5302e2158a254f25e4aba.

File: src/components/application_manager/test/CMakeLists.txt (hunk 15 above)
Comment (wolfylambova22) [inline src/components/application_manager/test/CMakeLists.txt]: @lea255ace  fixed in ba0bc031e6ecb39a14d5302e2158a254f25e4aba.
